builder/uploader/jenkins_upload.py

```diff
@@ -1,16 +1,37 @@
 import requests
+from functools import partial
 
 headers = {'Content-type': 'text/xml'}
 
 
+def krb_request(request_fn, config):
+    response = None
+    try:
+        from requests_kerberos import HTTPKerberosAuth, DISABLED
+        # mutual auth disabled because
+        # https://github.com/requests/requests-kerberos/issues/54
+        response = request_fn(auth=HTTPKerberosAuth(
+            mutual_authentication=DISABLED))
+        # if something goes wrong with requests_kerberos (eg. the issue
+        # linked above) an exception is not thrown but requests.post
+        # returns None object
+        if ((response is None) or
+           (response is not None and response.status_code == 401)):
+            response = request_fn(auth=(config['user'],
+                                        config['password']))
+    except ImportError:
+        response = request_fn(auth=(config['user'],
+                                    config['password']))
+    return response
+
+
 def post(url, payload, config):
-    response = requests.post(url,
-                             data=payload,
-                             headers=headers,
-                             auth=(config['user'],
-                                   config['password']),
-                             verify=False)
-    if response.status_code != 200:
+    do_post = partial(requests.post, url,
+                      data=payload,
+                      headers=headers,
+                      verify=False)
+    response = krb_request(do_post, config)
+    if response is not None and response.status_code != 200:
```

Comment:
Not sure if requests.get or requests.post might or might not None so I played it safe.